pkg/lib/cockpit-components-form-helper.tsx

```diff
@@ -22,7 +22,15 @@ import React from "react";
 import { FormHelperText } from "@patternfly/react-core/dist/esm/components/Form/index.js";
 import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/components/HelperText";
 
-export const FormHelper = ({ helperText, helperTextInvalid, variant, icon, fieldId }) => {
+export const FormHelper = ({ helperText, helperTextInvalid, variant, icon, fieldId } :
+  {
+      helperText?: string,
+      helperTextInvalid?: string,
```

Comment:
It looks like helperText and helperTextInvalid are required? Well either helperText or helperTextInvalid is which is a horrible API. Lets see if TypeScript can help us here.

Reply:
Seems it does support "signature" overloading but not sure how one fits that into this situation.

Reply:
Which it seems you can achieve by typing as:

https://blog.logrocket.com/implementing-function-overloading-typescript/

Reply:
Ya.  I made them both optional because it's possible for either one of them to be missing.  We could *probably* do something like:

```typescript
  {
      helperText?: string,
      helperTextInvalid?: string,
      ...
  } & ({ helperText: string } | { helperTextInvalid: string })
```

but I'm not sure if that will actually be an improvement or not — it seems more likely to get in the way of someone wanting to use it in a more dynamic way (for example, with two variables, each of which might potentially be `undefined` — the user "knows" that one of them is set, but typescript is bad at correlating type constraints over separate variables, so it would probably not be convinced).

All-in-all, I think it probably makes sense to leave it as it is.

Reply:
I honestly think this is the most bogus component in pkg/lib and I don't like it at all.

Reply:
And undefined is passed all the time..

```
+                <FormHelper fieldId="groups-create-name" helperTextInvalid={errors?.name} />
```

Reply:
In fact, I've found a number of users like this:

```jsx
    <FormHelper fieldId="accounts-create-real-name" helperTextInvalid={errors?.real_name} />
```

Where it's clear that the `helperTextInvalid` is potentially undefined and there is no `helperText` present.  In that case, no banner is shown:

```typescript
    if (!(helperText || helperTextInvalid))
        return null;
```

We might insist that at least one attribute is provided, albeit with a possibly-`undefined` value (ie: difference between `?: string` and `: string | undefined`)...